src/main/webapp/js/script.js

```diff
@@ -0,0 +1,270 @@
+ * video with a given `url`, and send data to the backend servlet
+ * @param url - a url for a YouTube video
+ * @requires - an authenticated user
+ */
+// Make sure the client is loaded and sign-in is complete before calling this method.
+function execute(url) {
+    // user inputs YouTube video URL
+    var videoId = "";
+
+    try {
+        videoId = getIdFromUrl(url);
+    } catch {
+        renderYtError("Invalid youtube url!");
+        return;
+    }
+
+    player = new YT.Player('player', {
+        height: '390',
+        width: '640',
+        videoId: videoId,
+        events: {'onReady': onPlayerReady, 'onStateChange': onPlayerStateChange}
+    });
+
+    return gapi.client.youtube.captions.list({
+      "videoId": videoId,
+      "part": [
+        "id"
+      ]
+    }).then(function(response) {
+        if (response.result != null & response.result.items != null &
+            response.result.items.length > 0) {
+            const trackId = response.result.items[0].id;
+
+            getCaptions(trackId, url).then(json => {
+                // send to backend
+                console.log("final destination", json);
+                sendJsonForm(json);
+            });
+        }
+    }, function(err) {
+        // top level error handler
+        console.error("Execute error", err); 
+    });
+}
+
+/**
+ * Create and send a form with the captions JSON. May need 
+ * to authenticate users here too to prevent malicious requests.
+ * @param json - the JSON string representing the 
+ *               parsed captions response
+ */
+function sendJsonForm(json) {
+    var params = new URLSearchParams();
+    params.append('json', json);
+    fetch('/caption', {
+            method: 'POST',
+            body: params,
+        }).then((response) => response.json()).then((json) => {
+            var output = '';
+            var numCap = 0;
+            var time = 0;
+            for (var key in json) {
+                output += key + ': ' + JSON.stringify(json[key]) + '<br>';
+            }
+            //$('#nlp-output').html(output);
+            document.getElementById('output').innerHTML = output;
+            alert(output);
+            numCap = json['METADATA'][0];
+            time = json['METADATA'][1];
+            //alert('Number of Captions: ' + numCap + '\nExecution Time: ' + time);
+        });
+}
+
+/************************************************* */
+/*        JS for Video Player Manipulation         */
+/************************************************* */
+
+// This code loads the IFrame Player API code asynchronously.
+var tag = document.createElement('script');
+
+tag.src = 'https://www.youtube.com/iframe_api';
+var firstScriptTag = document.getElementsByTagName('script')[0];
+firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+
+// The API will call this function when the video player is ready.
+function onPlayerReady(event) {
+  event.target.playVideo();
+}
```

Comment:
Maybe add comments for method headers for this, as well as the methods below this one.